### DISCUSSION TITLE
Linux headers question

### DISCUSSION THREAD

**Discussion by ghost** [dslm4515/CMLFS#36] [General]
When building and installing the Linux headers this guide uses these series of commands (similar to LFS).
> ARCH=${CMLFS_ARCH} make headers
> ...
> rm -v /cgnutools/${CMLFS_TARGET}/include/Makefile

However, when I install headers I usually use the command.
> make headers_install ARCH=${CMFLS_ARCH} INSTALL_HDR_PATH=/cgnutools/${CMLFS_TARGET}

From a quick check I don't see much difference between the two. Is there any difference (I'm a bit of a noob here :P)?

**Comment by andzai1995** [comment]
make headers_install requires rsync. When you make cross-compiler, you can use rsync because it is installed on your host system. On later stage (when you enter chroot) you can not execute rsync due to its absense. That's why we use this more complicated way in all three "builds" of linux-headers.

**Reply by ghost** [reply]
Ah, that makes sense.
Better to not include too much dependencies. :)
